include/aspect/velocity_boundary_conditions/interface.h

```diff
@@ -59,12 +59,23 @@ namespace aspect
          * Initialization function. This function is called once at the
          * beginning of the program after parse_parameters is run and after
          * the SimulatorAccess (if applicable) is initialized.
+         *
+         * @depricated Use initialize(types::boundary_id boundary_id) instead.
          */
         virtual
         void
         initialize ();
```

Comment:
You can use the `DEAL_II_DEPRECATED` macro here, as in
`initialize () DEAL_II_DEPRECATED;` This way every plugin that still uses this function will throw a warning at compile-time. You would then want to go through all plugins (and eventually all benchmarks and tests as well) that use this function and replace them with the new version (even if they just ignore the parameter, as in: `initialize(types::boundary_id /*boundary_id*/)` so that we have a master branch that does not throw hundreds of warnings. The deprecation is mostly for other people (like me) who have written their own plugins, to let them know they have to switch to the new formulation eventually. I will make the change gladly by the way, I have suffered from this limitation, and just did not had the time to make the change :+1: for fixing this.